### PR TITLE
Recommend against using defaults in Production

### PR DESCRIPTION
**Pull request by jpsamaroo**
While `Mux.defaults` is great for development and debugging purposes, it
should definitely not be used in a Production application, due to the
inclusion of the `basiccatch` middleware, which dumps stacktraces to
clients. Reword the README to indicate that one should instead roll
their own stack.

### CODE REVIEW

**Comment by aviks** [comment]
Sensible suggestion, but can we provide some recommendations  on what to do, rather than just what not to do? What is our recommendation on the best way to use this in production?

**Comment by jpsamaroo** [comment]
That's a great question! I haven't yet run Mux in "production" (still working towards that day), so if you have any concrete ideas I'd be happy to add them. I'll think on this a bit myself.

**Comment by jpsamaroo** [comment]
We might also want to include another middleware that is the same as `defaults` that replaces `basiccatch` with console or file logging. I can implement that.

**Comment by jpsamaroo** [comment]
@aviks updated per my previous suggestion, we now have `Mux.prod_defaults` which swaps `basiccatch` for `stderrcatch`. Confirmed to not dump sensitive backtraces to clients on my local machine.

**Comment by jpsamaroo** [comment]
Bump

File: src/basics.jl [hunk 1]
```diff
@@ -92,3 +92,12 @@ function basiccatch(app, req)
     return d(:status => 500, :body => codeunits(String(take!(io))))
   end
 end
+
+function stderrcatch(app, req)
+  try
+    app(req)
+  catch e
+    showerror(stderr, e, catch_backtrace())
```
Comment (aviks) [inline src/basics.jl]: Should this use Base's logging facility? Maybe that is a wider question.

**Pull request merged**